Solutions/Corvus.Monitoring.Instrumentation.Abstractions.Specs/SourceTaggingExceptionsInstrumentationSpecs.cs

```diff
@@ -28,10 +28,10 @@ public void WhenExceptionWithNoAdditionalInfoIsSentItShouldIncludeSource()
             Assert.AreSame(ex2, ex2Detail.Exception, "Exception (2)");
 
             Assert.AreEqual(1, ex1Detail.AdditionalDetail?.Properties.Count, "Property count (1)");
-            Assert.AreEqual(typeof(TestType1).FullName, ex1Detail.AdditionalDetail.Properties[this.Context.SourcePropertyName], "Source (1)");
+            Assert.AreEqual(typeof(TestType1).FullName, ex1Detail.AdditionalDetail?.Properties[this.Context.SourcePropertyName], "Source (1)");
```

Comment:
A minor point (and probably not worth updating the PR for) but I'd probably have used `!` here. The expectation is that `AdditionalDetail` will be non-null, and `!` is the way to express that. `?` expresses the expectation that it might be null, and it actually changes the type of the resulting expression to be nullable.

Normally, using `?` when `!` would be more appropriate goes on to cause other warnings. Here you've got away with it because `Assert.AreEqual` is prepared to accept null arguments, so the change in expression type doesn't cause additional warnings. So in a sense it doesn't really matter. But I'd prefer `!` because it says what you mean.

Reply:
I think I'd choose to make this change.

Reply:
Updated 👍